File: frontend/src/api/kafka.js
import api from './index.js'

export const getKafkaClusters = (params) => api.get('/kafka/clusters', { params })
export const getKafkaCluster = (id) => api.get(`/kafka/clusters/${id}`)
export const createKafkaCluster = (data) => api.post('/kafka/clusters', data)
export const updateKafkaCluster = (id, data) => api.put(`/kafka/clusters/${id}`, data)
export const deleteKafkaCluster = (id) => api.delete(`/kafka/clusters/${id}`)
export const testKafkaCluster = (id) => api.post(`/kafka/clusters/${id}/test`)
export const getKafkaClusterOptions = () => api.get('/kafka/clusters/options')

export const scanKafkaNetwork = (data) => api.post('/kafka/discovery/scan', data)
export const probeKafkaBootstrapServers = (data) => api.post('/kafka/discovery/probe', data)
export const importKafkaDiscoveryResult = (data, config = {}) => api.post('/kafka/discovery/import', data, config)

export const createKafkaTopic = (data) => api.post('/kafka/topics', data)
export const getKafkaTopics = (params) => api.get('/kafka/topics', { params })
export const deleteKafkaTopic = (clusterId, topic) => api.delete(`/kafka/topics/${encodeURIComponent(topic)}`, { params: { clusterId } })
export const updateKafkaTopicConfig = (topic, data) => api.put(`/kafka/topics/${encodeURIComponent(topic)}/config`, data)
export const getKafkaTopicPartitions = (clusterId, topic) => api.get(`/kafka/topics/${encodeURIComponent(topic)}/partitions`, { params: { clusterId } })
export const increaseKafkaTopicPartitions = (topic, data) => api.post(`/kafka/topics/${encodeURIComponent(topic)}/partitions`, data)

export const getKafkaBrokers = (clusterId) => api.get('/kafka/brokers', { params: { clusterId } })
export const updateKafkaBrokerConfig = (brokerId, clusterId, data) =>
  api.put(`/kafka/brokers/${encodeURIComponent(brokerId)}/config`, data, { params: { clusterId } })

export const getKafkaConsumerGroups = (params) => api.get('/kafka/consumer-groups', { params })
export const getKafkaConsumerGroupDetail = (groupId, params) => api.get(`/kafka/consumer-groups/${encodeURIComponent(groupId)}`, { params })
export const deleteKafkaConsumerGroup = (groupId, clusterId) => api.delete(`/kafka/consumer-groups/${encodeURIComponent(groupId)}`, { params: { clusterId } })
export const resetKafkaGroupOffset = (groupId, data) => api.post(`/kafka/consumer-groups/${encodeURIComponent(groupId)}/reset-offset`, data)

export const getKafkaMessages = (params) => api.get('/kafka/messages', { params })
export const produceKafkaMessage = (data) => api.post('/kafka/messages/produce', data)

export const getKafkaAuditLogs = (params) => api.get('/kafka/audit-logs', { params })

File: backend/internal/routes/kafka/kafka.go
package kafka

import (
	kafkactrl "devops-console-backend/internal/controllers/kafka"

	"github.com/gin-gonic/gin"
)

func RegisterKafkaRouters(apiGroup *gin.RouterGroup) {
	controller := kafkactrl.NewController()
	kafkaGroup := apiGroup.Group("/kafka")
	{
		clusterGroup := kafkaGroup.Group("/clusters")
		{
			clusterGroup.GET("", controller.ListClusters)
			clusterGroup.GET("/options", controller.ListClusterOptions)
			clusterGroup.GET("/:id", controller.GetCluster)
			clusterGroup.POST("", controller.CreateCluster)
			clusterGroup.PUT("/:id", controller.UpdateCluster)
			clusterGroup.DELETE("/:id", controller.DeleteCluster)
			clusterGroup.POST("/:id/test", controller.TestClusterConnection)
		}

		kafkaGroup.GET("/dashboard", controller.GetDashboard)
		kafkaGroup.POST("/topics", controller.CreateTopic)
		kafkaGroup.GET("/topics", controller.ListTopics)
		kafkaGroup.DELETE("/topics/:topic", controller.DeleteTopic)
		kafkaGroup.PUT("/topics/:topic/config", controller.UpdateTopicConfig)
		kafkaGroup.GET("/topics/:topic/partitions", controller.GetTopicPartitions)
		kafkaGroup.POST("/topics/:topic/partitions", controller.IncreaseTopicPartitions)
		kafkaGroup.GET("/brokers", controller.ListBrokers)
		kafkaGroup.PUT("/brokers/:id/config", controller.UpdateBrokerConfig)
		kafkaGroup.GET("/consumer-groups", controller.ListConsumerGroups)
		kafkaGroup.GET("/consumer-groups/:groupId", controller.GetConsumerGroupDetail)
		kafkaGroup.DELETE("/consumer-groups/:groupId", controller.DeleteConsumerGroup)
		kafkaGroup.POST("/consumer-groups/:groupId/reset-offset", controller.ResetConsumerGroupOffset)
		kafkaGroup.GET("/messages", controller.BrowseMessages)
		kafkaGroup.POST("/messages/produce", controller.ProduceMessage)
		kafkaGroup.GET("/audit-logs", controller.ListAuditLogs)
		kafkaGroup.POST("/discovery/scan", controller.ScanKafkaNetwork)
		kafkaGroup.POST("/discovery/probe", controller.ProbeKafkaBootstrapServers)
		kafkaGroup.POST("/discovery/import", controller.ImportDiscoveredKafka)
	}
}
